lib/ansible/plugins/lookup/ini.py

```diff
@@ -108,36 +118,36 @@ def get_value(self, key, section, dflt, is_regexp):
 
     def run(self, terms, variables=None, **kwargs):
 
+        self.set_options(var_options=variables, direct=kwargs)
+        paramvals = self.get_options()
+
         self.cp = configparser.ConfigParser()
 
         ret = []
         for term in terms:
-            params = _parse_params(term)
-            key = params[0]
-
-            paramvals = {
-                'file': 'ansible.ini',
-                're': False,
-                'default': None,
-                'section': "global",
-                'type': "ini",
-                'encoding': 'utf-8',
-            }
 
+            key = term
             # parameters specified?
-            try:
-                for param in params[1:]:
-                    name, value = param.split('=')
-                    if name not in paramvals:
-                        raise AnsibleAssertionError('%s not in paramvals' %
-                                                    name)
-                    paramvals[name] = value
-            except (ValueError, AssertionError) as e:
-                raise AnsibleError(e)
-
+            if '=' in term or ' ' in term.strip():
+                self._deprecate_inline_kv()
+                params = _parse_params(term, paramvals)
+                try:
+                    for param in params:
+                        if '=' in param:
+                            name, value = param.split('=')
+                            if name not in paramvals:
+                                raise AnsibleLookupError('%s is not a valid option.' % name)
+                            paramvals[name] = value
+                        elif key == term:
+                            # only take first, this format never supported multiple keys inline
```

Comment:
The comment is right, but there's a little change in behavior where something like
```
- debug: msg="{{ lookup('ini', 'file=users.ini', 'user', section='integration') }}"
```
```
# users.ini
[integration]
user=test
```
which also returns the value of 'user' with a comma.
```
ok: [localhost] => {
    "msg": ",test"
}
```

Reply:
hmm, i get a templating error
```
fatal: [localhost]: FAILED! => {
    "msg": "template error while templating string: invalid syntax for function call expression. String: User in integration is {{ lookup('ini', file='users.ini', 'user', section='integration') }}"
}
```

Reply:
ah, no diff quoting, but still, i get error:
```
fatal: [localhost]: FAILED! => {
    "msg": "An unhandled exception occurred while running the lookup plugin 'ini'. Error was a <class 'ansible.errors.AnsibleParserError'>, original message: Invalid filename: 'None'"
}
```

Reply:
@bcoca the order matters too, did you copy/paste?

Reply:
yes, for the 2nd one